mysql/field.go

```diff
@@ -23,9 +23,25 @@ type Field struct {
 	DefaultValue       []byte
 }
 
-func (p FieldData) Parse() (f *Field, err error) {
-	f = new(Field)
+type FieldValueType uint8
+
+type FieldValue struct {
+	Type   FieldValueType
+	Uint64 uint64
+	Int64  int64
+	Float  float64
```

Comment:
We can merge these fields into one `data uint64`, and cast to the desired type when need.
For extreme optimization, we can even use the first 8 bytes of `String`'s slice header to store them.

Reply:
Current memory bottleneck is allocations over time, not a current memory usage. Such struct union may be (I'm not sure at all) helpful in far feature not now.

Reply:
I think we can use one uint64, because even we use different types, we still need to use FieldValueType to distinguish them and return the specific value.

Reply:
@bobotu, @siddontang, I changed FieldValue for your recommendation. But I think this complicates the code more than it saves memory :)